File: src/test/java/no/nav/bidrag/commons/CorrelationIdTest.java
package no.nav.bidrag.commons;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.*;

@DisplayName("CorrelationId")
class CorrelationIdTest {

  @Test
  @DisplayName("skal lage correlation id med eksisterende verdi")
  void skalLageCorrelationIdMedEksisterendeVerdi() {
    CorrelationId correlationId = CorrelationId.existing("eksisterende");

     assertThat(correlationId.get()).isEqualTo("eksisterende");
  }

  @Test
  @DisplayName("skal lage correlation id som tidsstemplet verdi")
  void skalLageCorrelationIdSomTidsstempletVerdi() {
    CorrelationId correlationId = CorrelationId.generateTimestamped("value");
    String timestamp = Long.toHexString(System.currentTimeMillis());

     assertThat(correlationId.get())
         .startsWith(timestamp.substring(0, timestamp.length() - 3))
         .contains("-value");
  }

  @Test
  @DisplayName("skal generere ny verdi når gitt Correlation ID er null")
  void skalGenerereNyVerdiNarGittVerdiErNull() {
    var correlationId = CorrelationId.existing(null);

    assertThat(correlationId.get()).contains("-correlationId");
  }
}

File: src/test/kotlin/no/nav/bidrag/commons/CorrelationIdTest.kt
package no.nav.bidrag.commons

import io.kotest.matchers.shouldBe
import io.kotest.matchers.string.shouldContain
import io.kotest.matchers.string.shouldStartWith
import no.nav.bidrag.commons.CorrelationId.Companion.existing
import no.nav.bidrag.commons.CorrelationId.Companion.generateTimestamped
import org.junit.jupiter.api.Test

internal class CorrelationIdTest {
    @Test
    fun `skal lage correlation id med eksisterende verdi`() {
        val correlationId = existing("eksisterende")
        correlationId.get() shouldBe "eksisterende"
    }

    @Test
    fun `skal lage correlation id som tidsstemplet verdi`() {
        val correlationId = generateTimestamped("value")
        val timestamp = java.lang.Long.toHexString(System.currentTimeMillis())
        correlationId.get() shouldStartWith timestamp.substring(0, timestamp.length - 3) shouldContain "-value"
    }

    @Test
    fun `skal generere ny verdi når gitt Correlation ID er null`() {
        val correlationId = existing(null)
        correlationId.get() shouldContain "-correlationId"
    }
}
